packages/cli/src/commands/generate.js

```diff
@@ -18,23 +23,19 @@ export const builder = (yargs) =>
       )}`
     )
 
+/** @type {Record<string, import('yargs').Options>} */
 export const yargsDefaults = {
   force: {
     alias: 'f',
     default: false,
     description: 'Overwrite existing files',
     type: 'boolean',
   },
-  javascript: {
-    alias: 'js',
-    default: true,
-    description: 'Generate JavaScript files',
-    type: 'boolean',
-  },
   typescript: {
     alias: 'ts',
-    default: false,
-    description: 'Generate TypeScript files',
+    default: project.isTypeScriptProject,
+    description:
+      'Generate TypeScript files. Enabled by default if we detect your project is typescript',
```

Comment:
> if we detect your project is typescript

Is that done in toml settings? Or inferred in some other fashion?

Reply:
Uses @redwoodjs/structure. Just checks for tsconfig.json files really!